### PR TITLE
86b1u133x

### PR DESCRIPTION
**Pull request by thlurte**
This PR includes various updates and refactoring across the project to improve authentication and authorization. Key changes involve modifications to the authentication process, middleware refactoring, and updates to the session management logic.

### Changes:

**Authentication Updates:**

- Added error handling and improved JWT signing process.

**Session Management:**

- Refactored `getSession` to handle server-side and client-side scenarios.
- Updated `JwtPayloadSession` class to get user id via JWT payload.

**Middleware Refactoring:**

Refactored to improve role-based access control (RBAC) and handle session retrieval more effectively.

**Component Updates:**

- Removed deprecated imports and fixed import issues.
- Updated the `Home` component to handle session role and display relevant information based on user role.
- Removed duplicate code and ensured consistency in session handling.

**Docker Configuration:**
- Updated configuration for PostgreSQL and pgAdmin services.
- Removed redundant `server/docker-compose.yml.`

**Removed Unused Files:**
- Deleted obsolete files such as `server/src/pages/mentor.tsx` to streamline the project.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -14,8 +14,6 @@ services:
       POSTGRES_MULTIPLE_DATABASES: test,postgres
     ports:
       - "5432:5432"
-    volumes:
-      - postgres:/var/lib/postgresql/data
```
Comment (bsaranga) [inline docker-compose.yml:18]: Why was this removed?

File: server/src/server_actions/getSession.ts [hunk 2]
```diff
@@ -1,43 +1,44 @@
...
+export async function getSession(reqCookies: RequestCookies | null = null): Promise<JwtPayloadSession> {
+    let payload: string | null = null;
 
-export { getSessionOnClient }
\ No newline at end of file
+    if (typeof window === "undefined") {
+        const sessionCookie = reqCookies ? reqCookies.get('token') : serverCookies().get('token');
+        payload = sessionCookie ? sessionCookie.value.split('.')[1] : null;
+    } else {
+        const cookies = document.cookie.split(';').find(c => c.trim().startsWith('token='));
```
Comment (bsaranga) [inline server/src/server_actions/getSession.ts]: Token shouldn't be accessible using document object model.

**Review by bsaranga — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (thlurte) [reply docker-compose.yml:18]: it caused some conflicts and it's already there few lines up ahead, during the last merge i mistakenly accepted both changes, so i had to remove one of them

File: server/src/server_actions/getSession.ts (hunk 2 above)
Comment (thlurte) [reply server/src/server_actions/getSession.ts]: ok

Comment (thlurte) [reply server/src/server_actions/getSession.ts]: are we very careful about this to prevent incidents like session hijacking?

**Comment by thlurte** [comment]
if there is a need for additional modifications, please let me know.

File: server/src/server_actions/getSession.ts (hunk 2 above)
Comment (bsaranga) [reply server/src/server_actions/getSession.ts]: The usual norm is to store session in cookies with the flags HTTP only and secure (transmitted only over HTTPS). By making cookies HTTP only, we're safe from XSS attacks. And by making them `secure` we're safe from traffic sniffing.

**Review by bsaranga — APPROVED** [review]
